src/web/src/getlog_GET.ecpp

```diff
@@ -279,238 +280,392 @@ class _clonedcode_Compressor
-        std::string content_filename = logname_base;
+        }
 
-        { /* Scope away some temporary variables */
+        std::string content_filename = logname_base;
+        {   // Scope away some temporary variables
             struct utsname u;
             time_t t;
 
             if (uname (&u) == -1) {
                 log_error ("uname() failed: '%s'", strerror (errno));
-            } else {
-                content_filename += "_";
-                content_filename += u.nodename;
+            }
+            else {
+                content_filename.append("_");
+                content_filename.append(u.nodename);
             }
 
             t = time(NULL);
             if ( t == (time_t)-1 ) {
                 log_error ("time(NULL) failed: '%s'", strerror (errno));
-            } else {
-                /* ASSUMPTION: time_t is an integer type (usually is long int)
-                 * or otherwise can convert to std::string */
+            }
+            else {
+                // ASSUMPTION: time_t is an integer type (usually is long int)
+                // or otherwise can convert to std::string
 
                 char buf[21] = {0x00};
                 int r = calendar_to_datetime(time(NULL), buf, 21);
                 if (r != 0) {
                     log_error ("Failed to convert timestamp");
-                } else {
+                }
+                else {
                     // replace colons ':' by dash '-'
-                    // prevents http://latkin.org/blog/2016/07/20/git-for-windows-accidentally-creates-ntfs-alternate-data-streams/
+                    // prevents http://latkin.org/blog/2016/07/20/
+                    //      git-for-windows-accidentally-creates-ntfs-alternate-data-streams/
                     // for our Windows users
-                    for (char* c = strchr (buf, ':');
-                        c != NULL;
-                        c = strchr (c, ':'))
-                    {
-                        *c = '-';
-                    }
+                    for (char* c = strchr(buf, ':'); c != NULL; c = strchr(c, ':'))
+                        { *c = '-'; c++; }
 
                     content_filename.append("_");
                     content_filename.append(buf);
                 }
             }
         }
 
-        if (logname_ext == "" || logname_ext == ".txt")
-            content_filename += ".txt";
-        else
-            content_filename += ".txt" + logname_ext; /* e.g. "messages*.txt.gz" */
+        bool delete_logfile = false; // for post cleanup
 
-        if (logname_ext == "" || logname_ext == ".txt") {
-            std::ifstream in( logfile.c_str() );
-            if (!in)
-                throw std::runtime_error(TRANSLATE_ME ("Could not open requested logfile: ") + logfile);
+        if (logname_base == "ALL") {
+            // make a tar archive (logfile) owning *all* logfiles
+            delete_logfile = true;
+
+            std::string tmpdir = "/tmp/IPM2-logs";
```

Comment:
so playing in the /tmp world in the end?!

Reply:
yep